### PR TITLE
feat: Support custom HTTP agents

### PR DESCRIPTION
**Pull request by alkihis**
Add support of custom HTTP agents. Correctly support timeout function of native `http` package.
Addresses #149.

### CODE REVIEW

**Comment by alkihis** [comment]
Todo: Documentation

Example usage
```ts
import * as HttpProxyAgent from 'http-proxy-agent';

// HTTP/HTTPS proxy to connect to
const proxy = process.env.HTTP_PROXY || 'http://1.1.1.1:3000';

// create an instance of the `HttpProxyAgent` class with the proxy server information
const agent = new HttpProxyAgent(proxy);

// Instanciate helper with the agent
const client = new TwitterApi('bearerToken', { agent });
```

File: src/types/client.types.ts [hunk 1]
```diff
@@ -60,3 +62,7 @@ export interface IClientTokenNone {
 }
 
 export type TClientTokens = IClientTokenNone | IClientTokenBearer | IClientTokenOauth | IClientTokenBasic | IClientOAuth2UserClient;
+
+export interface IClientSettings {
+  agent: Agent;
```
Comment (PLhery) [inline src/types/client.types.ts:67]: I'd prefer a more explicit userAgent ; otherwise works for me

**Review by PLhery — APPROVED** [review]

File: src/types/client.types.ts (hunk 1 above)
Comment (PLhery) [reply src/types/client.types.ts:67]: Oh sorry I missread

Comment (PLhery) [inline src/types/client.types.ts:67]: [optional] I would write httpAgent
(So people don't think it's related to the userAgent)

**Pull request merged**